query.go

```diff
@@ -53,15 +62,20 @@ func (dht *IpfsDHT) newQuery(k string, f queryFunc) *dhtQuery {
 	}
 }
 
+// QueryPathFunc is a function that holds a path's state in its closure.
+// It returns the QueryFunc for one particular path. Its parameter, `path`,
+// specifies the path index number from 0 to numPaths - 1
+type queryPathFunc func(path int) (qfunc QueryFunc)
```

Comment:
small nitpick: unnecessary naming of the return type
```suggestion
type queryPathFunc func(path int) QueryFunc
```
technically, you don't *need* to name the func parameter either unless you have multiple parameters, in which case you either have to name all of them or none of them but I wouldn't suggest removing the `path` name in this case, just a general knowledge comment :smile: